### DISCUSSION TITLE
416 (Requested Range Not Satisfiable)

### DISCUSSION THREAD

**Discussion by lonnylot** [minio/minio#17387] [Q&A] [answered]
When I try to request a video file to stream in the browser I get

![Screenshot 2023-06-08 at 5 08 28 PM](https://github.com/minio/minio/assets/656791/24e3383a-15e2-49fa-a3c1-508d3ebcf74b)

Does anyone have any idea how I can fix this?

**Comment by lonnylot** [comment]
I've discovered this is a bug in Minio.

The browser is making the request with the header:

`Range: bytes=0-`

According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range this is a valid header, but Minio is responding that it is invalid.

**Reply by ravindk89** [reply]
If you believe this is a bug, please open an issue. 

You must include the MinIO Version you are running and how you are testing this process.

If you can provide the `mc admin trace` for the server while this call happens it will help.

**Comment by klauspost** [comment, marked as answer]
@lonnylot Are you sure this is a bug? We [explicitly test](https://github.com/minio/minio/blob/1e39ca39c341b9e0bd1c801f7626ae516058e2f6/cmd/httprange_test.go#L30) that value.

Is your object 0 bytes? In that case it will not be invalid since the range request asks to include the first byte.

**Reply by lonnylot** [reply]
> @lonnylot Are you sure this is a bug? We [explicitly test](https://github.com/minio/minio/blob/1e39ca39c341b9e0bd1c801f7626ae516058e2f6/cmd/httprange_test.go#L30) that value.
> 
> Is your object 0 bytes? In that case it will not be invalid since the range request asks to include the first byte.

Hey @klauspost yeah that's what happened. I apologize for the noise. I hadn't realized, but the file was 0 bytes.

**Reply by klauspost** [reply]
Yeah. There are no valid range values for 0 byte files in the rfc.
